packages/testing-karma/default-config.js

```diff
@@ -1,6 +1,115 @@
-const createLegacyConfig = require('./legacy-config');
-const createEsmConfig = require('./esm-config');
+const path = require('path');
 
-const legacy = process.argv.find(arg => arg.includes('legacy'));
+const legacy = process.argv.find(arg => arg.includes('--legacy'));
+if (legacy) {
+  /* eslint-disable-next-line no-console */
+  console.warn(`testing-karma --legacy flag has been renamed to --compatibility`);
+}
 
-module.exports = legacy ? createLegacyConfig : createEsmConfig;
+const compatibility = legacy || process.argv.find(arg => arg.includes('--compatibility'));
+const coverage = process.argv.find(arg => arg.includes('--coverage'));
+const updateSnapshots = process.argv.find(arg => arg.includes('--update-snapshots'));
+const pruneSnapshots = process.argv.find(arg => arg.includes('--prune-snapshots'));
+
+/**
+ * Creates base karma configuration.
+ */
+module.exports = config => ({
+  browsers: ['ChromeHeadlessNoSandbox'],
+
+  files: ['__snapshots__/**/*.md'],
+
+  customLaunchers: {
+    ChromeHeadlessNoSandbox: {
+      base: 'ChromeHeadless',
+      flags: ['--no-sandbox', '--disable-setuid-sandbox'],
+    },
+  },
+
+  plugins: [
+    // resolve plugins relative to this config so that they don't always need to exist
+    // at the top level
+    require.resolve('@open-wc/karma-esm'),
+    require.resolve('karma-mocha'),
+    require.resolve('karma-mocha-reporter'),
+    require.resolve('karma-source-map-support'),
+    require.resolve('karma-coverage-istanbul-reporter'),
+    require.resolve('karma-snapshot'),
+    require.resolve('karma-mocha-snapshot'),
+    require.resolve('karma-chrome-launcher'),
+    require.resolve('./src/snapshot-filename-preprocessor.js'),
+
+    // fallback: resolve any karma- plugins
+    'karma-*',
+  ],
+
+  frameworks: ['esm', 'mocha', 'snapshot', 'mocha-snapshot', 'source-map-support'],
+
+  esm: {
+    coverage,
+    compatibility: compatibility ? 'all' : 'none',
```

Comment:
why only support "all" or "none" while the es-dev-server supports more?

Reply:
I was thinking that for tests the speed is a bit less important than for developing. The primary speed difference between compatibility modes is whether or not babel has to be used, which is very quickly the case for code coverage.

But we could also just make it a flag like es-dev-server, that does make it consistent.

Remember that this is just the default config, `karma-esm` supports all the flags and people could turn it on in their config.

Reply:
Made it an enum flag instead.